.circleci/config.yml

```diff
@@ -177,37 +195,31 @@ commands:
                       name: "Waiting for << parameters.wait >>"
                       command: riot -v run 'wait' << parameters.wait >>
 
-            - run:
-                command: "riot list --hash-only '<<parameters.pattern>>' | shuf | circleci tests split | xargs -n 1 riot -v run --exitfirst --pass-env -s"
+            - when:
+                condition:
+                  << pipeline.parameters.coverage >>
+                steps:
+                  - run:
+                      command: "riot list --hash-only '<<parameters.pattern>>' | shuf | circleci tests split | xargs -n 1 riot -v run --exitfirst --pass-env -s"
+            - unless:
+                condition:
+                  << pipeline.parameters.coverage >>
+                steps:
+                  - run:
+                      command: "riot list --hash-only '<<parameters.pattern>>' | shuf | circleci tests split | xargs -n 1 -I {} riot -v run --exitfirst --pass-env {} -s --no-cov"
       - save_pip_cache
       - when:
           condition:
-            << parameters.store_coverage >>
+            and:
+              - << pipeline.parameters.coverage >>
+              - << parameters.store_coverage >>
           steps:
             - save_coverage
       - store_test_results:
           path: test-results
       - store_artifacts:
           path: test-results
 
-  run_tox_scenario_with_testagent:
```

Comment:
This is no longer used and since we're migrating away from tox, I'm removing this.